clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7206,6 +7206,9 @@ ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D, const ParsedAttr &AL,
   case ParsedAttr::AT_EnumExtensibility:
     handleEnumExtensibilityAttr(S, D, AL);
     break;
+  case ParsedAttr::AT_SYCLExternal:
+    S.SYCL().handleExternalAttr(D, AL);
```

Comment:
I think `handleSimpleAttribute()` can be called here and the diagnostics handled elsewhere. See additional comments in `SemaSYCL::handleExternalAttr()`.